moda-contracts/src/interfaces/ICatalog.sol

```diff
@@ -0,0 +1,336 @@
+     * @param releases - The address of the Releases contract
+     */
+    function getReleasesOwner(address releases) external view returns (address owner);
+
+    /**
+     * @dev Add or revoke permission for a Releases contract to release a track
+     * @param trackId - The track id of the track
+     * @param releases - The address of the Releases contract
+     * @param hasPermission - The permission to set true/false
+     */
+    function grantReleasesAccessSingle(
+        string calldata trackId,
+        address releases,
+        bool hasPermission
+    ) external;
+
+    /**
+     * @dev Sets permission for a Releases contract to be able to release
+     * all tracks owned by the caller
+     * @param artist - The address of the artist
+     * @param releases - The address of the Releases contract
+     * @param hasPermission - The permission to set true/false
+     */
+    function grantReleasesAccessAll(address artist, address releases, bool hasPermission) external;
+
+    /**
+     * @dev Returns true/false if a Releases contract has permission
+     * to release a track
+     * @param trackId - The track id of the track
+     * @param releases - The address of the Releases contract
+     */
+    function getReleasesAccessSingle(
+        string calldata trackId,
+        address releases
+    ) external view returns (bool);
+
+    /**
+     * @dev Returns true/false if a Releases contract has permission
+     * to release all tracks owned by the caller
+     * @param artist - The address of the artist
+     * @param releases - The address of the Releases contract
+     */
+    function getReleasesAccessAll(address artist, address releases) external view returns (bool);
+
+    /**
+     * @dev Registers a release. In order for a release to be registered
+     * the Releases contract must be registered, the tracks must be registered,
+     * the Releases contract must have the track owners permission to release the track,
+     * and this exact combination of ordered tracks and metadata uri must not have been
+     * registered previously.
+     * @param trackIds - The track ids of the tracks
+     * @param uri - The metadata uri of the release
+     * @param tokenId - The token id of the release
+     */
+    function registerRelease(
+        string[] calldata trackIds,
+        string calldata uri,
+        uint256 tokenId
+    ) external;
+
+    /**
+     * @dev Unregisters a releases. This deletes a release hash enabling the release
+     * to be created again.
+     * @notice Only the default admin can call this function.
+     * @param releaseHash - The hash of the release
+     */
+    function unregisterRelease(bytes32 releaseHash) external;
+
+    /**
+     * @dev Checks the caller is the track owner or manager
+     * @param trackId The id of the track
+     * @param caller The address of the caller
+     */
+    function requireUserIsTrackArtistOrManager(string calldata trackId, address caller) external view;
```

Comment:
Is this used anywhere? This seems like it should return a bool, not be the one to revert if another contract is calling it.

`hasTrackAccess(trackId, account) => bool` seems like a better name. The caller can then choose to revert, or do something else

Reply:
Oh yep my bad it should return a bool, this will be used by the releases contract when it is open, to check it the user that is creating the release owns the tracks or is a manager, since with an open releases contract the user won't be granting access to the tracks like they would if it was closed.

Reply:
I think isTrackArtistOrManager as a function name works in the context of running a check when creating a release as those are the only entities that can do so right?

Reply:
If it is private then it can revert, but it is not a good idea to do so if it is public